triton-vm/src/table/lookup_table.rs

```diff
@@ -0,0 +1,263 @@
+
+        let mut constraints = [
+            lookup_input_is_0,
+            cascade_table_log_derivative_is_initialized_correctly,
+            public_evaluation_argument_is_initialized_correctly,
+        ];
+        ConstraintCircuitMonad::constant_folding(&mut constraints);
+        constraints.map(|circuit| circuit.consume()).to_vec()
+    }
+
+    pub fn ext_consistency_constraints_as_circuits() -> Vec<ConstraintCircuit<SingleRowIndicator>> {
+        vec![]
+    }
+
+    pub fn ext_transition_constraints_as_circuits() -> Vec<ConstraintCircuit<DualRowIndicator>> {
+        let circuit_builder = ConstraintCircuitBuilder::new();
+        let one = circuit_builder.b_constant(BFIELD_ONE);
+        let one_montgomery = circuit_builder.b_constant(BFieldElement::from_raw_u64(1));
+
+        let current_base_row = |col_id: LookupBaseTableColumn| {
+            circuit_builder.input(CurrentBaseRow(col_id.master_base_table_index()))
+        };
+        let next_base_row = |col_id: LookupBaseTableColumn| {
+            circuit_builder.input(NextBaseRow(col_id.master_base_table_index()))
+        };
+        let current_ext_row = |col_id: LookupExtTableColumn| {
+            circuit_builder.input(CurrentExtRow(col_id.master_ext_table_index()))
+        };
+        let next_ext_row = |col_id: LookupExtTableColumn| {
+            circuit_builder.input(NextExtRow(col_id.master_ext_table_index()))
+        };
+        let challenge = |challenge_id: ChallengeId| circuit_builder.challenge(challenge_id);
+
+        let lookup_input = current_base_row(LookIn);
+        let is_padding = current_base_row(IsPadding);
+        let cascade_table_server_log_derivative = current_ext_row(CascadeTableServerLogDerivative);
+        let public_evaluation_argument = current_ext_row(PublicEvaluationArgument);
+
+        let lookup_input_next = next_base_row(LookIn);
+        let lookup_output_next = next_base_row(LookOut);
+        let lookup_multiplicity_next = next_base_row(LookupMultiplicity);
+        let is_padding_next = next_base_row(IsPadding);
+        let cascade_table_server_log_derivative_next =
+            next_ext_row(CascadeTableServerLogDerivative);
+        let public_evaluation_argument_next = next_ext_row(PublicEvaluationArgument);
+
+        // Padding section is contiguous: if the current row is a padding row, then the next row
+        // is also a padding row.
+        let if_current_row_is_padding_row_then_next_row_is_padding_row =
+            is_padding * (one.clone() - is_padding_next.clone());
+
+        // Lookup Table's input increments by 1 if and only if the next row is not a padding row
+        let if_next_row_is_padding_row_then_lookup_input_next_is_0 =
+            is_padding_next.clone() * lookup_input_next.clone();
+        let if_next_row_is_not_padding_row_then_lookup_input_next_increments_by_1 = (one.clone()
+            - is_padding_next.clone())
+            * (lookup_input_next.clone() - lookup_input - one_montgomery);
```

Comment:
@aszepieniec Can this constraint be expressed without the need to explicitly refer to Montgomery representation? Variable `one_montgomery` is defined as:

```rust
let one_montgomery = circuit_builder.b_constant(BFieldElement::from_raw_u64(1));
```